libsolidity/formal/SMTChecker.cpp

```diff
@@ -1310,6 +1340,26 @@ void SMTChecker::resetVariables(function<bool(VariableDeclaration const&)> const
 	});
 }
 
+bool SMTChecker::equalNonValueTypes(VariableDeclaration const& _var1, VariableDeclaration const& _var2)
+{
+	if (!_var1.hasReferenceOrMappingType() || !_var2.hasReferenceOrMappingType())
+		return false;
+	TypePointer type1 = _var1.type();
+	TypePointer type2 = _var2.type();
+	if (type1->category() == Type::Category::Mapping || type2->category() == Type::Category::Mapping)
+		return *type1 == *type2;
+	auto refType1 = dynamic_cast<ReferenceType const*>(type1.get());
```

Comment:
Can you explain why you have to jump through these hoops here?

Reply:
This function was simplified.